### ISSUE TITLE
Slack Alert notification not being sent

### ISSUE THREAD

**Issue by thechile** [grafana/grafana#15972]
Using `v5.4.2` i have setup a Slack Notification Channel.

URL: `valid hooks url`
Recipient: `@user.name`

I click the `Send Test` button and i receive the test **successfully**. I also see `lvl=info msg="Executing slack notification" logger=alerting.notifier.slack ` in the logs.

I than try to use this same notification channel within a panel alert and when i click the test button it tells me `alerting` but nothing happens. I receive nothing in the grafana server logs either. I added `alerting.notifier.slack:debug` into the config but it doesn't seem to help.

My understanding is that `alerting` means "something should be sent".

Is there something else i need to do or reason why Slack alerts might go AWOL ?

Thanks.

**Comment by thechile** [comment]
It doesn't seem to be just Slack, if i try with a webhook alert it also goes AWOL.

When i configure a dummy localhost webhook alert and use the test i get something:

```
POST / HTTP/1.1
Host: localhost:9999
User-Agent: Grafana
Content-Length: 337
Content-Type: application/json
Accept-Encoding: gzip

{"evalMatches":[{"value":100,"metric":"High value","tags":null},{"value":200,"metric":"Higher Value","tags":null}],"message":"Someone is testing the alert notification within grafana.","ruleId":0,"ruleName":"Test notification","ruleUrl":"https://grafana/","state":"alerting","title":"[Alerting] Test notification"}
```

but when i use it for a panel alert, nothing even though it says alerting.

**Comment by thechile** [comment]
ok i see. But given that the only purpose for the alert rule is to trigger a notification then if there is a test button i'd expect it to send a test notification when i click it. @torkelo 

To me this behavior is not obvious and the behavior is flawed.

**Comment by torkelo** [comment]
Test rule is to test the rule evaluation not the notification channel (it has a separate test feature).

But understand the confusion, if we get more feedback with similar thoughts I will agree that it’s flawed, but this is the first of this kind :)

**Comment by pawansinha0612** [comment]
Can someone please help here - How to get notification to slack. ?
I can just see "added an integration to this channel: notification"
 After that alerting happens in grafana but no notification is not coming to slack channel

**Comment by kangelos** [comment]
I am using rocket.chat which is more or less slack compatible, the error I am seeing is "Invalid href" so my guess is that the alert page url as passed from grafana is not accepted from slack/rocket.chat 
Here is a sample log:

> Integrations ➔ Incoming WebHook.debug [Process Incoming Request result of Trigger Grafana :]
> Integrations ➔ Incoming WebHook.debug result { attachments: 
>    [ { color: '#D63232',
>        fallback: '[Alerting] Uptime alert',
>        fields: [Array],
>        footer: 'Grafana v6.4.3',
>        footer_icon: 'https://grafana.com/assets/img/fav32.png',
>        image_url: '',
>        text: '@all ',
>        title: '[Alerting] Uptime alert',
>        title_link: 'https://grafana.mydomain.io/d/cpLF04gZk/panel-name?fullscreen&edit&tab=alert&panelId=29&orgId=1',
>        ts: 1571821621 } ] }
> API ➔ debug Failure { statusCode: 400,
>   body: 
>    { success: false,
>      error: 'Invalid href value provided',
>      stack: undefined } }

**Comment by benj0c** [comment]
I had problem that test alert works (send notification alert on slack), but it doesn't work when alert occurs on dashboard. The problem was in notification channel properties include image (it was enabled) and I didn't configure image rendering.

My proposal is to make that test alert also fail if image rendering doesn't configure. It will make more sense, and helps people to quickly find a lack.

**Comment by bryannaegele** [comment]
I can confirm what @benj0c found. Image rendering works fine for us in test alerts but failed on real ones. If image rendering fails, the alert should still always be sent

**Comment by zmailloux** [comment]
Also confirming @benj0c 's findings, real annoyance given how little the Grafana logs tell us.

**Comment by marefr** [comment]
#21536 included in v6.6.0 should make sure the alert is sent even though image rendering fails.

**Comment by wutzebaer** [comment]
Maybe the docs should mention that we have to add an notification to the alert

![grafik](https://user-images.githubusercontent.com/10062396/112069396-0a45af80-8b6c-11eb-901d-cec00994e641.png)
